src/parsers/containers/isobmff/take_pssh_out.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import log from "../../../log";
+import { bytesToHex } from "../../../utils/byte_parsing";
+import {
+  getBoxContent,
+  getBoxOffsets,
+} from "./get_box";
+
+export interface IPSSHInfo { systemID : string;
+                             data : Uint8Array; }
+
+/**
+ * Replace every PSSH box from an ISOBMFF segment by FREE boxes and returns the
+ * removed PSSH in an array.
+ * Useful to manually manage encryption while avoiding the round-trip with the
+ * browser's encrypted event.
+ * @param {Uint8Array} data - the ISOBMFF segment
+ * @returns {Array.<Uint8Array>} - The extracted PSSH boxes. In the order they
+ * are encountered.
+ */
+export default function takePSSHOut(data : Uint8Array) : IPSSHInfo[] {
```

Comment:
I was thinking. Could it have an interest to reduce the size of PSSH by turning it into a hashmap ?